aim/sdk/adapters/xgboost.py

```diff
@@ -0,0 +1,45 @@
+from xgboost.callback import TrainingCallback, CallbackContainer
+from aim.sdk.session.session import Session
+from typing import Optional
+
+class AimCallback(TrainingCallback):
+    def __init__(self, repo: Optional[str] = None,
+                 experiment: Optional[str] = None):
+        super().__init__()
+        self.repo = repo
+        self.experiment = experiment
+        self.initialized = False
+
+    def before_training(self, model):
+        self.aim_session = Session(
+            repo=self.repo,
+            experiment=self.experiment
+        )
+        self.initialized = True
+        return model
+
+    def after_iteration(self, model, epoch: int,
+                        evals_log: CallbackContainer.EvalsLog) -> bool:
+        if not evals_log:
+            return False
+
+        for data, metric in evals_log.items():
+            for metric_name, log in metric.items():
```

Comment:
Do metric names have prefixes that define the training phase? e.g. `train_`, `val_`. If yes, then lets automatically set corresponding `subset` context.

Reply:
there were no such prefixes, no `train_begin` or `val_begin` functions as well. I couldn't find a way to get the current phase.